include\fpm\ios.hpp only works from C++14

According to the readme fpm runs from C++11 or higher.
If the include file "include\fpm\ios.hpp" is included, the gcc compiler reports:

```
ios.hpp:645:68: error: use of 'auto' in lambda parameter declaration only available with -std=c++14 or -std=gnu++14
         if (std::all_of(significand.begin(), significand.end(), [](auto x){ return x == 0; })) {
```
C++11 doesn't support generic lambdas. 

Suggestion: Note in the readme that output via ostream only works from C++14 or higher.
(Or replace the lambda with a functor to be compatible with C++11)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,7 +30,7 @@ add_executable(fpm-test
   tests/power.cpp
   tests/trigonometry.cpp
 )
-set_target_properties(fpm-test PROPERTIES CXX_STANDARD 14)
+set_target_properties(fpm-test PROPERTIES CXX_STANDARD 11)
 target_link_libraries(fpm-test PRIVATE fpm gtest_main)
 gtest_add_tests(TARGET fpm-test)
 

--- a/include/fpm/ios.hpp
+++ b/include/fpm/ios.hpp
@@ -642,7 +642,7 @@ std::basic_istream<CharT, Traits>& operator>>(std::basic_istream<CharT, Traits>&
     // We've parsed all we need. Construct the value.
     if (exponent_overflow) {
         // Absolute exponent is too large
-        if (std::all_of(significand.begin(), significand.end(), [](auto x){ return x == 0; })) {
+        if (std::all_of(significand.begin(), significand.end(), [](unsigned char x){ return x == 0; })) {
             // Significand is zero. Exponent doesn't matter.
             x = fixed<B, I, F>(0);
         } else if (exponent_negate) {

--- a/tests/customizations.cpp
+++ b/tests/customizations.cpp
@@ -51,8 +51,8 @@ struct Limits<fpm::fixed_16_16>
     static constexpr int max_exponent() noexcept { return  15; }
     static constexpr int min_exponent10() noexcept { return -4; }
     static constexpr int max_exponent10() noexcept { return 4; }
-    static constexpr auto min() noexcept { return fpm::fixed_16_16::from_raw_value(-2147483647 - 1); }
-    static constexpr auto max() noexcept { return fpm::fixed_16_16::from_raw_value( 2147483647); }
+    static constexpr fpm::fixed_16_16 min() noexcept { return fpm::fixed_16_16::from_raw_value(-2147483647 - 1); }
+    static constexpr fpm::fixed_16_16 max() noexcept { return fpm::fixed_16_16::from_raw_value( 2147483647); }
 };
 
 template <>
@@ -65,8 +65,8 @@ struct Limits<fpm::fixed_24_8>
     static constexpr int max_exponent() noexcept { return 23; }
     static constexpr int min_exponent10() noexcept { return -2; }
     static constexpr int max_exponent10() noexcept { return 6; }
-    static constexpr auto min() noexcept { return fpm::fixed_24_8::from_raw_value(-2147483647 - 1); }
-    static constexpr auto max() noexcept { return fpm::fixed_24_8::from_raw_value( 2147483647); }
+    static constexpr fpm::fixed_24_8 min() noexcept { return fpm::fixed_24_8::from_raw_value(-2147483647 - 1); }
+    static constexpr fpm::fixed_24_8 max() noexcept { return fpm::fixed_24_8::from_raw_value( 2147483647); }
 };
 
 template <>
@@ -79,8 +79,8 @@ struct Limits<fpm::fixed_8_24>
     static constexpr int max_exponent() noexcept { return  7; }
     static constexpr int min_exponent10() noexcept { return -7; }
     static constexpr int max_exponent10() noexcept { return  2; }
-    static constexpr auto min() noexcept { return fpm::fixed_8_24::from_raw_value(-2147483647 - 1); }
-    static constexpr auto max() noexcept { return fpm::fixed_8_24::from_raw_value( 2147483647); }
+    static constexpr fpm::fixed_8_24 min() noexcept { return fpm::fixed_8_24::from_raw_value(-2147483647 - 1); }
+    static constexpr fpm::fixed_8_24 max() noexcept { return fpm::fixed_8_24::from_raw_value( 2147483647); }
 };
 
 TYPED_TEST(customizations, numeric_limits)

--- a/tests/output.cpp
+++ b/tests/output.cpp
@@ -106,20 +106,12 @@ private:
     std::stringstream create_stream() const
     {
         std::stringstream ss;
-        ss.setf(combine_or(GetParam(), std::make_index_sequence<4>{}));
+        ss.setf(std::get<0>(GetParam()) | std::get<1>(GetParam()) | std::get<2>(GetParam()) | std::get<3>(GetParam()));
         ss.precision(std::get<4>(GetParam()));
         ss.width(std::get<5>(GetParam()));
         ss.fill(std::get<6>(GetParam()));
         ss.imbue(std::get<7>(GetParam()));
         return ss;
-    }
-
-    template <std::size_t... I>
-    static fmtflags combine_or(const Flags& flags, std::index_sequence<I...>)
-    {
-        fmtflags result{};
-        std::make_tuple((result |= std::get<I>(flags))...);
-        return result;
     }
 };
 

